js/index.js

```diff
@@ -0,0 +1,325 @@
+    for (let i = 0; i < productList.length; i++) {
+        if (productList[i].productid === productid || productList[i].productname.toLowerCase() === productname.toLowerCase()) {
+            swal("Enter unique Product Id And Name!", {
+                icon: "warning",
+                timer: 2000
+            });
+            return false;
+        }
+    }
+    return true;
+}
+
+//Function to validate Name while editing data
+function validateName(index) {
+    var productname = document.getElementById("productname").value;
+    var productList;
+    productList = (localStorage.getItem("productList") == null) ? productList = [] : JSON.parse(localStorage.getItem("productList"));
+    for (let i = 0; i < productList.length; i++) {
+        if (productList[i].productname.toLowerCase() === productname.toLowerCase()) {
+            if (productList[index].productname.toLowerCase() === productname.toLowerCase()) {
+                return true;
+            }
+            else {
+                swal("Enter unique Product Name!", {
+                    icon: "warning",
+                    timer: 2000
+                });
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+//Function to clear data after using modal
+function clearValue() {
+    $('#addAdmin').modal('hide');
+    document.getElementById("productid").value = "";
+    document.getElementById("productname").value = "";
+    document.getElementById("image").value = "";
+    document.getElementById("imgedit").src = "";
+    document.getElementById("price").value = "";
+    document.getElementById("description").value = "";
+}
+
+//Function to sort/filter
+function changeBy(x) {
+    var productList;
+    productList = (localStorage.getItem("productList") == null) ? productList = [] : JSON.parse(localStorage.getItem("productList"));
+    switch (x) {
+        case "lowFirst":
+            productList.sort((a, b) => {
+                return a.price - b.price;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "highFirst":
+            productList.sort((a, b) => {
+                return b.price - a.price;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "idFirst":
+            productList.sort((a, b) => {
+                return a.productid - b.productid;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "nameFirst":
+            productList.sort(function (a, b) {
+                var x = a.productname.toLowerCase();
+                var y = b.productname.toLowerCase();
+                return x < y ? -1 : x > y ? 1 : 0;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
```

Comment:
Why this repeat in every case?
Is that possible you can pass that in parameter in to showData function and then set in local storage?

Reply:
resolve this line by passing parameter productList in new function which set items.
function setItems(setProductItems){
    localStorage.setItem('productList', JSON.stringify(setProductItems));
}
setItems(productList);